### ISSUE TITLE
sequence_parallel_size设置大于1时，是一定要pack_to_max_length = True吗？

### ISSUE THREAD

**Issue by fengstar7827** [InternLM/xtuner#997]
我在进行full sft qwen32B，发现只要sequence_parallel_size设置大于1时，pack_to_max_length = False时就会报错：

Assertion `idx_dim >= 0 && idx_dim < index_size && "index out of bounds"` failed

目前我的config：
 Model
pretrained_model_name_or_path = "/DeepSeek-R1-Distill-Qwen-32B"
use_varlen_attn = False

# # parallel
sequence_parallel_size = 8

# Data
data_files = ["sample_18k_xtuner.jsonl"]
prompt_template = PROMPT_TEMPLATE.qwen_chat
max_length = 61440  # 60k
pack_to_max_length = False
# max_position_embeddings = 61440

# Scheduler & Optimizer
batch_size = 1  # per_device
accumulative_counts = 8 # bs = 1 GPU * 1 batch_size_per_device * 16 acc
accumulative_counts *= sequence_parallel_size
dataloader_num_workers = 4
max_epochs = 10
optim_type = AdamW
lr = 2e-5
betas = (0.9, 0.999)
weight_decay = 0
max_norm = 1  # grad clip
warmup_ratio = 0.03
log_interval = 1

tokenizer = dict(
    type=AutoTokenizer.from_pretrained,
    pretrained_model_name_or_path=pretrained_model_name_or_path,
    trust_remote_code=True,
    padding_side="right",
    eos_token='<|im_end|>',
)

model = dict(
    type=SupervisedFinetune,
    use_varlen_attn=use_varlen_attn,
    llm=dict(
        type=AutoModelForCausalLM.from_pretrained,
        # max_position_embeddings=max_position_embeddings,
        pretrained_model_name_or_path=pretrained_model_name_or_path,
        trust_remote_code=True,
        torch_dtype=torch.float16,
        # quantization_config=dict(
        #     type=BitsAndBytesConfig,
        #     load_in_4bit=True,
        #     load_in_8bit=False,
        #     llm_int8_threshold=6.0,
        #     llm_int8_has_fp16_weight=False,
        #     bnb_4bit_compute_dtype=torch.float16,
        #     bnb_4bit_use_double_quant=True,
        #     bnb_4bit_quant_type="nf4",
        # ),
    ),
    # lora=dict(
    #     type=LoraConfig,
    #     r=64,
    #     lora_alpha=64,
    #     lora_dropout=0.1,
    #     bias="none",
    #     task_type="CAUSAL_LM",
    # ),
)

**Comment by fengstar7827** [comment]
升级tranformers后解决

**Comment by 2Elian** [comment]
> 升级变形金刚后解决

请问升级到哪个版本？我最近尝试使用SP的时候，也会出现这个错误，我觉得是SP采样器的BUG
